### PR TITLE
Fix a few minor schema processing bugs

### PR DESCRIPTION
**Pull request by kgaillot**
... and simplify update_validation()

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens, here's the next batch of schema code updates

File: lib/common/tests/schemas/pcmk__cmp_schemas_by_name_test.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                                          "pacemaker-1.0") > 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-1.2",
+                                          "pacemaker-2.0") < 0);
+}
+
+PCMK__UNIT_TEST(setup, teardown,
+                cmocka_unit_test(unknown_is_lesser),
+                cmocka_unit_test(none_is_greater),
+                cmocka_unit_test(next_is_before_none),
+                cmocka_unit_test(known_numeric));
```
Comment (clumens) [inline lib/common/tests/schemas/pcmk__cmp_schemas_by_name_test.c:98]: It'd be worth adding a test here and there that uses uppercase in the schema names.  It looks like pcmk__get_schema uses case-insensitive comparisons.

Comment (kgaillot) [reply lib/common/tests/schemas/pcmk__cmp_schemas_by_name_test.c]: added commit on top

**Review by clumens — APPROVED** [review]

**Comment by kgaillot** [comment]
CI failures are due to libqb not found on a couple of builders, appears to be unrelated infrastructure issue

**Comment by kgaillot** [comment]
retest this please

**Pull request merged**